source/source_stats_reporter.go

```diff
@@ -98,6 +106,15 @@ func (r *reporter) ReportEventCount(args *ReportArgs, responseCode int) error {
 	return nil
 }
 
+func (r *reporter) ReportRetryEventCount(args *ReportArgs, responseCode int) error {
+	ctx, err := r.generateTag(args, responseCode)
+	if err != nil {
+		return err
+	}
+	metrics.Record(ctx, retryEventCountM.M(1))
```

Comment:
retry can happen more than once per attempt. I have not seen how this is intended to be used

Reply:
you're right.. it happens more than once per attempt.. so the related PR: https://github.com/knative/eventing/pull/4861 is submitted in eventing repo.. that will call this reporter for every retry and that's why I am adding count here.

Reply:
@n3wscott does that help?

Reply:
Yeah, that makes sense to me. Otherwise you'd have to pass in something like an array of structs since if you collate the retry count with all the responseCodes that caused the failure, that seems wrong? Or, maybe I misunderstand what the question here was.